File: src/test/java/mc/sbm/simphonycloud/domain/FactureEnTeteTestSamples.java
package mc.sbm.simphonycloud.domain;

import java.util.Random;
import java.util.UUID;
import java.util.concurrent.atomic.AtomicInteger;

public class FactureEnTeteTestSamples {

    private static final Random random = new Random();
    private static final AtomicInteger intCount = new AtomicInteger(random.nextInt() + (2 * Short.MAX_VALUE));

    public static FactureEnTete getFactureEnTeteSample1() {
        return new FactureEnTete()
            .id(1)
            .num(1)
            .factureRef("factureRef1")
            .nbrePax(1)
            .numTable(1)
            .taxeMontantTotal(1)
            .sousTotal(1)
            .factureTotal(1)
            .commissionTotal(1)
            .tipTotal(1)
            .remiseTotal(1)
            .erreursCorrigeesTotal(1)
            .retourTotal(1)
            .xferToFactureEnTeteRef(1)
            .xferStatus("xferStatus1")
            .categoriePointDeVenteRef(1)
            .pointDeVenteRef(1);
    }

    public static FactureEnTete getFactureEnTeteSample2() {
        return new FactureEnTete()
            .id(2)
            .num(2)
            .factureRef("factureRef2")
            .nbrePax(2)
            .numTable(2)
            .taxeMontantTotal(2)
            .sousTotal(2)
            .factureTotal(2)
            .commissionTotal(2)
            .tipTotal(2)
            .remiseTotal(2)
            .erreursCorrigeesTotal(2)
            .retourTotal(2)
            .xferToFactureEnTeteRef(2)
            .xferStatus("xferStatus2")
            .categoriePointDeVenteRef(2)
            .pointDeVenteRef(2);
    }

    public static FactureEnTete getFactureEnTeteRandomSampleGenerator() {
        return new FactureEnTete()
            .id(intCount.incrementAndGet())
            .num(intCount.incrementAndGet())
            .factureRef(UUID.randomUUID().toString())
            .nbrePax(intCount.incrementAndGet())
            .numTable(intCount.incrementAndGet())
            .taxeMontantTotal(intCount.incrementAndGet())
            .sousTotal(intCount.incrementAndGet())
            .factureTotal(intCount.incrementAndGet())
            .commissionTotal(intCount.incrementAndGet())
            .tipTotal(intCount.incrementAndGet())
            .remiseTotal(intCount.incrementAndGet())
            .erreursCorrigeesTotal(intCount.incrementAndGet())
            .retourTotal(intCount.incrementAndGet())
            .xferToFactureEnTeteRef(intCount.incrementAndGet())
            .xferStatus(UUID.randomUUID().toString())
            .categoriePointDeVenteRef(intCount.incrementAndGet())
            .pointDeVenteRef(intCount.incrementAndGet());
    }
}

File: src/main/webapp/app/entities/facture-en-tete/facture-en-tete.test-samples.ts
import dayjs from 'dayjs/esm';

import { IFactureEnTete, NewFactureEnTete } from './facture-en-tete.model';

export const sampleWithRequiredData: IFactureEnTete = {
  id: 25171,
};

export const sampleWithPartialData: IFactureEnTete = {
  id: 682,
  num: 3202,
  ouvertureDateTime: dayjs('2025-05-26T04:07'),
  nbrePax: 9349,
  factureTotal: 9434,
  commissionTotal: 23583,
  tipTotal: 15192,
  xferStatus: 'requite',
  categoriePointDeVenteRef: 24866,
  pointDeVenteRef: 18362,
};

export const sampleWithFullData: IFactureEnTete = {
  id: 25992,
  num: 17366,
  factureRef: 'ew',
  ouvertureDateTime: dayjs('2025-05-25T18:54'),
  fermetureDateTime: dayjs('2025-05-26T12:54'),
  estAnnule: false,
  nbrePax: 25866,
  numTable: 27116,
  taxeMontantTotal: 10491,
  sousTotal: 5722,
  factureTotal: 24077,
  commissionTotal: 10903,
  tipTotal: 4527,
  remiseTotal: 8389,
  erreursCorrigeesTotal: 21523,
  retourTotal: 672,
  xferToFactureEnTeteRef: 28347,
  xferStatus: 'unlike',
  categoriePointDeVenteRef: 29125,
  pointDeVenteRef: 2691,
};

export const sampleWithNewData: NewFactureEnTete = {
  id: null,
};

Object.freeze(sampleWithNewData);
Object.freeze(sampleWithRequiredData);
Object.freeze(sampleWithPartialData);
Object.freeze(sampleWithFullData);
